src/services/compare/handlers/sendDoesNotExistAlert.ts

```diff
@@ -20,47 +27,152 @@ exports.handler = async function (
 
-      // you can also use the data.programType value here if needed "MAC" | "HHS" | "CHP"
-      const params = getRecordDoesNotExistParams({
-        emailRecipients,
-        sourceEmail,
-        transmittalNumber,
+      // i have defined these variables to use them as flags to fill the email data fields dynamocally
+      let recipientType;
+      let recipients;
+      const isChp = data.programType == "CHP"; // isChp caintains true||false
+
+      let emailData = { sourceEmail: emailParams.sourceEmail };
+
+      // depending on if chip true||false filling in the info
+      if (isChp) {
+        const { CHP } = emailParams;
+        // CHP contains emailRecipientsInitial, emailRecipientsFirstFollowUp, emailRecipientsSecondFollowUp
+        const {
+          emailRecipientsInitial,
+          emailRecipientsFirstFollowUp,
+          emailRecipientsSecondFollowUp,
+        } = CHP;
+        emailData["emailRecipientsInitial"] = emailRecipientsInitial;
+        emailData["emailRecipientsFirstFollowUp"] =
+          emailRecipientsFirstFollowUp;
+        emailData["emailRecipientsSecondFollowUp"] =
+          emailRecipientsSecondFollowUp;
+      }
+      {
+        const { nonCHP } = emailParams;
+        const {
+          emailRecipientsInitial,
+          emailRecipientsFirstFollowUp,
+          emailRecipientsSecondFollowUp,
+        } = nonCHP;
+        emailData["emailRecipientsInitial"] = emailRecipientsInitial;
+        emailData["emailRecipientsFirstFollowUp"] =
+          emailRecipientsFirstFollowUp;
+        emailData["emailRecipientsSecondFollowUp"] =
+          emailRecipientsSecondFollowUp;
+      }
+
+      // defining an object to map and calculate init, first and second followup email
+      const emailRecipientsTypes = {
+        emailRecipientsInitial:
+          !(data.secSinceMmdlSigned > 48 * 2 * 3600) &&
+          !(
+            data.secSinceMmdlSigned > 48 * 3600 &&
+            data.secSinceMmdlSigned < 48 * 2 * 3600
+          ),
+        emailRecipientsFirstFollowUp:
+          data.secSinceMmdlSigned > 48 * 3600 &&
+          data.secSinceMmdlSigned < 48 * 2 * 3600,
+        emailRecipientsSecondFollowUp: data.secSinceMmdlSigned > 48 * 2 * 3600,
+      };
+
+      // if it greater then 2 days but less then 4 days
+      if (emailRecipientsTypes.emailRecipientsFirstFollowUp) {
+        recipientType = "emailRecipientsFirstFollowUp";
+        recipients = emailData["emailRecipientsFirstFollowUp"];
+      }
+      // if it is greater then 4 days
+      else if (emailRecipientsTypes.emailRecipientsSecondFollowUp) {
+        recipientType = "emailRecipientsSecondFollowUp";
+        recipients = emailData["emailRecipientsSecondFollowUp"];
+      }
+      // if it is less then 2 days
+      else if (emailRecipientsTypes.emailRecipientsInitial) {
+        recipientType = "emailRecipientsInitial";
+        recipients = emailData["emailRecipientsInitial"];
+      }
```

Comment:
this was a wild ride to read. lets sync up tomorrow and see if we can find a better way.